res/skins/QMLDemo/Deck.qml

```diff
@@ -0,0 +1,263 @@
+import Mixxx 0.1 as Mixxx
+import Mixxx.Controls 0.1 as MixxxControls
+import QtQuick 2.15
+import QtQuick.Controls 2.15
+import QtQuick.Layouts 1.11
+
+Item {
```

Comment:
This is already quite big, I'd prefer to split everything into its own file if possible. At least Items which are already decoupled. Accessing properties from the deck itself could either be done by making use of [QMLs dynamic scoping](https://doc.qt.io/qt-5/qtqml-documents-scope.html) or by using (alias) properties.

Reply:
Hmm, I'd rather not make too many changes to not cause too many conflicts with the follow-up QML PRs. This is just a demo skin anyway, so we can refactor later :D It's more important that the reusable components in the `Mixxx` subdirectory are well-structured.

Reply:
As long as it is a Demo skin thats fine, but the first QML skin should be well polished and using QML best practices. Otherwise others will copy that mediocre code and we end up with a similar situation as with the mappings.

Reply:
Yes. I hope that once the demo skin is merged, other people start working on it and improve it, until we end up with a nice, polished skin.